Backend/TimeTracker.Api/Controllers/TimerController.cs

```diff
@@ -0,0 +1,216 @@
+
+            Timer timer = await database.Timers
+                .AsNoTracking()
+                .Where(t => t.User.Id == authHelper.GetCurrentUserId(User))
+                .Include(t => t.Project)
+                .FirstOrDefaultAsync(t => t.Id == timerId);
+
+            if(timer == null) {
+                return new GenericResponseDTO<TimerDTO> {
+                    Message = "Could not find timer with a matching Id.",
+                    Success = false
+                };
+            }
+
+            return new GenericResponseDTO<TimerDTO> {
+                Data = new TimerDTO{
+                    Id = timer.Id,
+                    StartTime = timer.StartTime,
+                    Notes = timer.Notes,
+                    ProjectId = timer.Project.Id 
+                },
+                Success = true
+            };
+            
+        }
+
+        // returns a list of timerDTOs for each timer belonging to the logged in user
+        [HttpGet]
+        public async Task<GenericResponseDTO<List<TimerDTO>>> GetAllTimers() {
+
+            int currentUserId;
+
+            try {
+                currentUserId = authHelper.GetCurrentUserId(User);
+            } catch(System.NullReferenceException) {
+                return new GenericResponseDTO<List<TimerDTO>> {
+                    Message = "Not logged in.",
+                    Success = false
+                };
+            }
+
+            List<Timer> timers = await database.Timers
+                .AsNoTracking()
+                .Where(t => t.User.Id == currentUserId)
+                .Include(t => t.Project)
+                .ToListAsync();
+
+            List<TimerDTO> timerDTOs = new List<TimerDTO>();
+            timers.ForEach(t => timerDTOs.Add(new TimerDTO {
+                Id = t.Id,
+                StartTime = t.StartTime,
+                Notes = t.Notes,
+                ProjectId = t.Project.Id
+            }));
+
+            return new GenericResponseDTO<List<TimerDTO>> {
+                Data = timerDTOs,
+                Success = true
+            };
+        }
+
+        // returns a list of timerDTOs for each timer belonging to the logged in user
+        [HttpGet]
+        [Route("DateRange")]
+        public async Task<GenericResponseDTO<List<TimerDTO>>> GetTimersWithinDateRange(DateTime startDate, DateTime endDate) {
+
+            int currentUserId;
+
+            try {
+                currentUserId = authHelper.GetCurrentUserId(User);
+            } catch(System.NullReferenceException) {
+                return new GenericResponseDTO<List<TimerDTO>> {
+                    Message = "Not logged in.",
+                    Success = false
+                };
+            }
+
+            List<Timer> timers = await database.Timers
+                .AsNoTracking()
+                .Where(t => t.User.Id == currentUserId && t.StartTime > startDate && endDate > t.StartTime)
+                .Include(t => t.Project)
+                .ToListAsync();
```

Comment:
Here just set the start date to make sure it's at the very start of the day and the end date is at the very end of the day. We have done this in a few other spots.